stash_engine/lib/stash/link_out/labslink_service.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require 'net/ftp'
+
+require_relative 'helper'
+
+module LinkOut
+
+  # Dryad also furnishes links to Europe PMC via the LabsLink service. They support a
+  # similar mechanism to NCBI LinkOut, where we provide a list of PMIDs and destination
+  # data packages.
+  #   Europe PMC LabsLink: http://europepmc.org/LabsLink
+  class LabslinkService
+
+    include ::LinkOut::Helper
+
+    attr_reader :provider_file
+    attr_reader :links_file
+
+    def initialize
+      @ftp = APP_CONFIG.link_out.labslink
+      @schema = 'http://europepmc.org/docs/labslink.xsd'
+      @links_file = 'labslink-links.xml'
+      @doi_url_stem = 'http://dx.doi.org/'
+      @provider_file = 'labslink-profile.xml'
+      @root_url = Rails.application.routes.url_helpers.root_url.freeze
+    end
+
+    def generate_files!
+      p "  created #{generate_provider_file!}"
```

Comment:
Hi Brian,

I'm not sure what context these are run in, but it may make sense to log instead of puts.  I can't say I understand all the context very well.  Looking forward to see the demo tomorrow so I understand it all better.

Reply:
ayiyi. yes I meant to go back and do that. It was all originally run from a rake task and I was having trouble with the Rails logger.